pkg/ipamd/datastore/data_store.go

```diff
@@ -145,38 +148,57 @@ type ENI struct {
 	IsTrunk bool
 	// IsEFA indicates whether this ENI is tagged as an EFA
 	IsEFA bool
-	// IsPDEnabled indicates whether prefix delegation is enabled
-	IsPDEnabled bool
 	// DeviceNumber is the device number of ENI (0 means the primary ENI)
 	DeviceNumber int
 	// IPv4Addresses shows whether each address is assigned, the key is IP address, which must
 	// be in dot-decimal notation with no leading zeros and no whitespace(eg: "10.1.0.253")
-	IPv4Addresses map[string]*AddressInfo
-	// Key is the prefix and value is prefix, used IPs under the prefix
-	IPv4Prefixes map[string]*ENIPrefix
+	// Key is the IP address
+	AvailableIPv4Cidrs map[string]*AssignedIPv4Addresses
```

Comment:
it's not clear to me what Key is reading this comment for a CIDR.

Reply:
I can update the comment, Key will be "192.0.2.1/24"

Reply:
is this Key every used? if not it's better to be []*AssignedIPv4Addresses

Reply:
Yes mainly in delete of the prefix while IPs are not used or during reducing the datastore, we can directly index rather than walking another loop.